activesupport/lib/active_support/cache.rb

```diff
@@ -700,7 +694,7 @@ def normalize_version(key, options = nil)
         def expanded_version(key)
           case
           when key.respond_to?(:cache_version) then key.cache_version.to_param
-          when key.is_a?(Array)                then key.map { |element| expanded_version(element) }.compact.to_param
+          when key.is_a?(Array)                then key.map { |element| expanded_version(element) }.compact!.to_param
```

Comment:
`compact!` returns `nil` when there are no changes made, which feels like it would be an issue here

Reply:
Ah, good catch. I fixed if to use `tap(&:compact!)` so that it doesn't return nil.